R/between-within.R

```diff
@@ -0,0 +1,105 @@
+#' Calculation of Between-Within Degrees of Freedom for One-Dimensional Contrast
+#'
+#' @description Calculates the estimate, standard error, degrees of freedom,
+#' t statistic and p-value for one-dimensional contrast. Used in [df_1d()] if method is
+#' "Between-within".
+#'
+#' @param object (`mmrm`)\cr the MMRM fit.
+#' @param contrast (`numeric`)\cr contrast vector. Note that this should not include
+#'   elements for singular coefficient estimates, i.e. only refer to the
+#'   actually estimated coefficients.
+#'
+#' @return List with `est`, `se`, `df`, `t_stat` and `p_val`.
+#' @keywords internal
+h_df_1d_bw <- function(object, contrast) {
+  assert_class(object, "mmrm")
```

Comment:
lots of duplicate code here, between the two functions but also with other functions. Todo for Daniel: DRY